libs/utils/src/tracing_span_assert.rs

```diff
@@ -252,13 +269,17 @@ mod tests {
     #[test]
     fn tracing_error_subscriber_not_set_up() {
         // no setup
-
         let span = tracing::info_span!("foo", e = "some value");
         let _guard = span.enter();
 
         let extractor = MultiNameExtractor::new("E", ["e"]);
-        let missing = check_fields_present([&extractor]).unwrap_err();
-        assert_missing(missing, vec![&extractor]);
+        check_fields_present([&extractor])
+            .expect("without any subscriber, should still return Ok(())");
+
+        // similarly for a not found key
+        let extractor = MultiNameExtractor::new("E", ["foobar"]);
+        check_fields_present([&extractor])
+            .expect("without any subscriber, should still return Ok(())");
```

Comment:
Hm, I kinda don't like that we could just drop the subscriber setup in a future pageserver refactoring, then not get alerted about it.

Since we're already relying on global state here, why not a global `bool` to control the behavior if no subscriber is set up?

Reply:
You mean a global in addition to what the `tracing` does already inside this file?

I do think the NoSubscriber is quite the strong indicator of tracing not being configured, which is why I added it.

Reply:
Split the two commits to #4682